Scripts/Interfaces/IUsesSnapping.cs

```diff
@@ -0,0 +1,62 @@
+﻿#if UNITY_EDITOR
+using System;
+using UnityEngine;
+
+namespace UnityEditor.Experimental.EditorVR
+{
+	/// <summary>
+	/// Gives decorated class the ability to use snapping
+	/// </summary>
+	public interface IUsesSnapping
+	{
+	}
+
+	public static class IUsesSnappingMethods
+	{
+		internal delegate bool ManipulatorSnappingDelegate(Transform rayOrigin, GameObject[] objects, ref Vector3 position, ref Quaternion rotation, Vector3 delta);
+		internal delegate bool DirectSnappingDelegate(Transform rayOrigin, GameObject go, ref Vector3 position, ref Quaternion rotation, Vector3 targetPosition, Quaternion targetRotation);
+
+		internal static ManipulatorSnappingDelegate manipulatorSnapping { get; set; }
```

Comment:
This should be a verb rather than a noun; manipulatorSnap? Adjust the delegate and external method, too